src/ServiceControl/Recoverability/Retries/RetryDocumentManager.cs

```diff
@@ -114,26 +114,33 @@ internal void AdoptOrphanedBatches(IDocumentSession session, out bool hasMoreWor
         {
             RavenQueryStatistics stats;
 
-            var orphanedBatchIds = session.Query<RetryBatch, RetryBatches_ByStatusAndSession>()
+            var orphanedBatches = session.Query<RetryBatch, RetryBatches_ByStatusAndSession>()
                 .Where(b => b.Status == RetryBatchStatus.MarkingDocuments && b.RetrySessionId != RetrySessionId)
                 .Statistics(out stats)
-                .Select(b => b.Id)
                 .ToArray();
 
-            log.InfoFormat("Found {0} orphaned retry batches from previous sessions", orphanedBatchIds.Length);
+            log.InfoFormat("Found {0} orphaned retry batches from previous sessions", orphanedBatches.Length);
 
-            AdoptBatches(session, orphanedBatchIds);
+            AdoptBatches(session, orphanedBatches.Select(b => b.Id));
+
+            foreach (var batch in orphanedBatches)
+            {
+                if (batch.RetryType != RetryType.MultipleMessages)
```

Comment:
We still need to load the retry type and the request id to mark the correct batch